lib/rubocop/rspec/example_group.rb

```diff
@@ -14,72 +14,44 @@ class ExampleGroup < Concept
         ExampleGroups::ALL + SharedGroups::ALL + Includes::ALL
       ).block_pattern
 
+      def lets
+        find_all_in_scope(node, :let?)
+      end
+
       def subjects
-        subjects_in_scope(node)
+        find_all_in_scope(node, :subject?)
       end
 
       def examples
-        examples_in_scope(node).map(&Example.public_method(:new))
+        find_all_in_scope(node, :example?).map(&Example.public_method(:new))
       end
 
       def hooks
-        hooks_in_scope(node).map(&Hook.public_method(:new))
+        find_all_in_scope(node, :hook?).map(&Hook.public_method(:new))
```

Comment:
```diff
- find_all_in_scope(node, :hook?).map(&Hook.public_method(:new))
+ find_all_in_scope(node, :hook?).map { |hook| Hook.new(hook) }
```
Looks even shorter when called with a block.
Are there any performance implications?

Reply:
The usual caveats about benchmarks apply, but it looks like the `&` syntax might actually be slower:

```
ampersand
  917906.0 i/s (±3.44%)
curlies
  1339367.1 i/s (±0.69%)

Comparison:
    curlies:   1339367.1 i/s
  ampersand:    917906.0 i/s - 1.46x slower
```

```rb
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'
  gem 'benchmark-inputs'
end

require "benchmark/inputs"

INPUTS = [
  [1, 2, 3],
  [1, nil, 2, nil, 3, nil],
  [nil, nil, nil],
]

class MyClass
  attr_accessor :foo
  def initialize(foo)
    self.foo = foo
  end
end

Benchmark.inputs(INPUTS) do |job|
  job.dup_inputs = true
  job.report("ampersand"){|array| array.map(&MyClass.method(:new)) }
  job.report("curlies"){|array| array.map { |element| MyClass.new(element) } }
  job.compare!
end
```

Reply:
TIL `benchmark-inputs` 👍 

Oh, I didn't know `&method` is slower, and couldn't even imagine that it's that slower.

Thanks for using the scientific approach here!